src/renderer/init.js

```diff
@@ -61,6 +65,20 @@ async function init() {
     connect,
   });
 
+  const oldUserId = global.localStorage.getItem("userId");
+  if (oldUserId) {
+    global.localStorage.removeItem("userId");
```

Comment:
imo this needs to be moved only after the get + set so make sure it's "atomic". otherwise if user kill the live before it ends being set, it would fail.

Reply:
https://github.com/LedgerHQ/ledger-live-desktop/pull/3888#discussion_r645625073 problem is we don't know when the setKey is actually done, there a debounce. or i change the setkey to be async or add additional callback ?
However i think its a very slight possibility that the user has time to close just in between

Reply:
so clearly to me this is a risk that worth a fix. we can't afford possibly losing the userId.